WcaOnRails/app/models/registration.rb

```diff
@@ -86,6 +83,10 @@ def best_solve(event, type)
     person ? person.best_solve(event, type) : SolveTime.new(event.id, type, 0)
   end
 
+  def event_picker_events
+    registration_events.map(&:event_object).sort_by(&:rank)
+  end
```

Comment:
Sorry, the deeper magic of the forms cycle is still not very clear to me.

I'm confident there is a "proper rails way" of doing this, so I'd Google around until I find either (1) how it's supposed to be done, or (2) the StackOverflow post describing how this is an idiotic bug the arrogant Rails people refuse to address.